internal/resources/credential/datasource_credential.go

```diff
@@ -39,14 +44,59 @@ func dataSourceCredentialRead(ctx context.Context, d *schema.ResourceData, m int
 
 	name, _ := d.Get(NameKey).(string)
 
-	resp, err = config.TMCConnection.CredentialResourceService.CredentialResourceServiceGet(constructFullname(d))
-	if err != nil || resp == nil {
-		if clienterrors.IsNotFoundError(err) && !helper.IsDataRead(ctx) {
-			_ = schema.RemoveFromState(d, m)
-			return diags
+	getCredentialResourceRetrybleFn := func() (retry bool, err error) {
+		resp, err = config.TMCConnection.CredentialResourceService.CredentialResourceServiceGet(constructFullname(d))
+		if err != nil || resp == nil {
+			if clienterrors.IsNotFoundError(err) && !helper.IsDataRead(ctx) {
+				_ = schema.RemoveFromState(d, m)
```

Comment:
Can we add clear comment on what is the case we remove the state? Like why only when it's any operation other than read with not found error?

This would make the code more readable and understandable.

Reply:
In case of data read even if the resource is not found we don't remove it from state, all other cases if resource is not found its removed from the state. This implementation remains the same across different resources in the repo and that's the reason same is followed here.I believe this is self explanatory, if we need to add comment then it has to be across all resources.

Reply:
Yes it is explanatory, still good to have a comment at least for this resource.